src/main/java/minijava/parser/Parser.java

```diff
@@ -319,16 +330,10 @@ private void parseExpressionWithPrecedenceClimbing(int minPrecedence) {
 
   /** UnaryExpression -> PostfixExpression | (! | -) UnaryExpression */
   private void parseUnaryExpression() {
-    switch (currentToken.terminal) {
-      case INVERT:
-      case MINUS:
-        consumeToken();
-        parseUnaryExpression();
-        break;
-      default:
-        parsePostfixExpression();
-        break;
+    while (currentToken.isOneOf(INVERT, MINUS)) {
+      consumeToken();
     }
+    parsePostfixExpression();
```

Comment:
Also eliminated this tail call, inspired by PR #28. Also made sure, that this was the last recursion we could eliminate, all other recursion doesn't happen in tail position.